MINOR: Use Topic::isInternalTopic instead of directly checking

soondenana opened the pull request:
We don't allow chaning number of partitions for internal topics. To do
so we check if the topic name belongs to the set of internal topics
directly instead of using the "isInternalTopic" method. This breaks the
encapsulation by making client aware of the fact that internal topics
have special names.

This is a simple change to use the method "Topic::isInternalTopic"
method instead of checking it directly in "alterTopic" command.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

soondenana committed:
MINOR: Use Topic::isInternalTopic instead of directly checking

We don't allow chaning number of partitions for internal topics. To do
so we check if the topic name belongs to the set of internal topics
directly instead of using the "isInternalTopic" method. This breaks the
encapsulation by making client aware of the fact that internal topics
have special names.

This is a simple change to use the method "Topic::isInternalTopic"
method instead of checking it directly in "alterTopic" command.

hachikuji commented on core/src/main/scala/kafka/admin/TopicCommand.scala:
```diff
@@ -333,7 +333,7 @@ object TopicCommand extends Logging {
         if(tp.hasPartitions) {
-          if (Topic.INTERNAL_TOPICS.contains(topic)) {
+          if (Topic.isInternal(topic)) {
             throw new IllegalArgumentException(s"The number of partitions for the internal topics${Topic.INTERNAL_TOPICS} cannot be changed.")
```
We may as well change the log message also. I don't see why it doesn't mention `topic` directly.

soondenana replied:
Done. Updated log comment. Also changed visibility of INTERNAL_TOPICS private so that in future clients only use Topic::isInternal method to check if a topic is internal.

soondenana committed:
Updated log message to print the topic for which alterTopic is called

soondenana committed:
Make INTERNAL_TOPICS private

Instead of using the variable directly, clients should use the
Topic::isInternal method.

hachikuji reviewed (approved):
LGTM
